docs/src/main/sphinx/develop/connectors.rst

```diff
@@ -79,33 +79,103 @@ features, like:
 Note that data modification also requires implementing
 a :ref:`connector-page-sink-provider`.
 
+When a ``SELECT`` query is being executed, and it contains specific clauses,
+the engine can try pushing down some of them into the connector, by calling
+one of the following methods of the ``ConnectorMetadata`` service:
+
+* ``applyLimit`` for the ``LIMIT`` clause, if ``ORDER BY`` is not present,
+  or ``applyTopN`` when ``ORDER BY`` is present
+* ``applyFilter`` for the ``WHERE`` clause
+* ``applyProjection`` for expressions in the ``SELECT`` clause
+* ``applySample`` for the ``TABLESAMPLE`` clause
+* ``applyAggregation`` for the ``GROUP BY`` clause
+* ``applyJoin`` for the different types of the ``JOIN`` clause
+* ``applyTableFunction`` for the ``TABLE ()`` table function clause
+* ``applyTableScanRedirect`` to redirect table scans to a different connector
```

Comment:
The applyXXX methods may or may not correspond to SQL clauses. They are related to operations in the plan IR, not directly with SQL language features. For instance, both the HAVING and WHERE clause get implemented as a Filter in the IR. In some cases, filters can be synthesized from other SQL constructs (e.g., expression in the JOIN clause could be pushed down as filters and eventually make it to connectors via applyFilter).

Therefore, it's more correct and accurate to describe these operations in terms of what plan structures they apply to.

Reply:
I reworded this section to mention the IR. I also removed these example clauses, so we could explain them better later in subsections as we do here for limit and top-N.